src/main/java/org/embulk/input/zendesk/ZendeskInputPluginDelegate.java

```diff
@@ -0,0 +1,532 @@
+                            builder.add(include, Types.JSON);
+                        }
+                    }
+                }
+            }
+            task.setSchema(targetSchemaBuilder.build());
+        }
+        else  {
+            List<Column> columns = task.getSchema().getColumns();
+            for (Column column : columns) {
+                if (column.getType() != Types.TIMESTAMP) {
+                    builder.add(column.getName(), column.getType());
+                }
+                else {
+                    builder.add(new JacksonTimestampValueLocator(column.getName()), column.getName(), column.getType(),
+                            ZendeskConstants.Misc.RUBY_TIMESTAMP_FORMAT);
+                }
+            }
+        }
+        return builder.build();
+    }
+
+    @VisibleForTesting
+    protected static ZendeskRestClient getZendeskV2RestClient(final PluginTask task)
+    {
+        int retryLimit = Exec.isPreview() ? 1 : task.getRetryLimit();
+        final Jetty92RetryHelper retryHelper = new Jetty92RetryHelper(retryLimit, task.getRetryInitialWaitSec() * 1000,
+                task.getMaxRetryWaitSec() * 1000,
+                new DefaultJetty92ClientCreator(task.getConnectionTimeout() * 1000, task.getConnectionTimeout() * 1000));
+        return new ZendeskRestClientImpl(retryHelper, task);
+    }
+
+    @VisibleForTesting
+    protected static ZendeskSupportAPiService getZendeskV2SupportAPiService(final ZendeskRestClient zendeskRestClient, final PluginTask task)
+    {
+        return new ZendeskSupportAPiServiceImpl(zendeskRestClient, task);
+    }
+
+    private boolean isLaterThanStartTime(String startTime, long time)
+    {
+        return time > ZendeskDateUtils.toTimeStamp(startTime);
+    }
+
+    private boolean isDuplicatedRecord(List<String> previousRecords, String recordId)
+    {
+        return previousRecords.contains(recordId);
+    }
+
+    private boolean isSupportInclude(Target target, List<String> includes)
+    {
+        return includes != null && !includes.isEmpty() && ZendeskUtils.isSupportInclude(target);
+    }
+
+    private void fetchData(final JsonNode jsonNode, final Target target, final List<String> includes,
+                           final RecordImporter recordImporter, final PageBuilder pageBuilder)
+    {
+        if (isSupportInclude(target, includes)) {
+            fetchRelatedObjects(jsonNode, target, includes, recordImporter, pageBuilder);
+        }
+        else {
+            recordImporter.importRecord(new JacksonServiceRecord((ObjectNode) getDataToImport(target, jsonNode)), pageBuilder);
+        }
+    }
+
+    private void fetchRelatedObjects(final JsonNode jsonNode, final Target target, final List<String> includes,
+                              final RecordImporter recordImporter, final PageBuilder pageBuilder)
+    {
+        // remove .json in url
+        String url = jsonNode.get("url").asText().trim();
```

Comment:
👇 A bit slower, but less error-prone:
```suggestion
        String url = jsonNode.get("url").asText().trim().replaceFirst("\\.json", "");
```